src/io/flutter/view/FlutterView.java

```diff
@@ -1,671 +1,687 @@
-  public static final String RENDER_TREE_LABEL = "Render Tree";
-
-  @NotNull
-  private final FlutterViewState state = new FlutterViewState();
-
-  @NotNull
-  private final Project myProject;
-
-  private String restoreToolWindowId;
-
-  private final Map<FlutterApp, PerAppState> perAppViewState = new HashMap<>();
-
-  public FlutterView(@NotNull Project project) {
-    myProject = project;
-  }
-
-  @Override
-  public void dispose() {
-  }
-
-  @NotNull
-  @Override
-  public FlutterViewState getState() {
-    return this.state;
-  }
-
-  @Override
-  public void loadState(FlutterViewState state) {
-    this.state.copyFrom(state);
-  }
+  /*
+   * Copyright 2017 The Chromium Authors. All rights reserved.
+   * Use of this source code is governed by a BSD-style license that can be
+   * found in the LICENSE file.
+   */
+  package io.flutter.view;
+
+  import com.intellij.execution.runners.ExecutionUtil;
+  import com.intellij.execution.ui.ConsoleViewContentType;
+  import com.intellij.execution.ui.layout.impl.JBRunnerTabs;
+  import com.intellij.icons.AllIcons;
+  import com.intellij.ide.BrowserUtil;
+  import com.intellij.ide.browsers.BrowserLauncher;
+  import com.intellij.openapi.Disposable;
+  import com.intellij.openapi.actionSystem.*;
+  import com.intellij.openapi.actionSystem.ex.CustomComponentAction;
+  import com.intellij.openapi.actionSystem.impl.ActionButton;
+  import com.intellij.openapi.application.ApplicationManager;
+  import com.intellij.openapi.components.PersistentStateComponent;
+  import com.intellij.openapi.components.Storage;
+  import com.intellij.openapi.project.Project;
+  import com.intellij.openapi.ui.SimpleToolWindowPanel;
+  import com.intellij.openapi.util.text.StringUtil;
+  import com.intellij.openapi.wm.ToolWindow;
+  import com.intellij.openapi.wm.ToolWindowManager;
+  import com.intellij.openapi.wm.ex.ToolWindowEx;
+  import com.intellij.openapi.wm.ex.ToolWindowManagerEx;
+  import com.intellij.ui.content.*;
+  import com.intellij.ui.*;
+  import com.intellij.ui.content.Content;
+  import com.intellij.ui.content.ContentFactory;
+  import com.intellij.ui.content.ContentManager;
+  import com.intellij.ui.tabs.TabInfo;
+  import icons.FlutterIcons;
+  import io.flutter.FlutterBundle;
+  import io.flutter.FlutterInitializer;
+  import io.flutter.inspector.InspectorService;
+  import io.flutter.run.daemon.FlutterApp;
+  import io.flutter.run.daemon.FlutterDevice;
+  import io.flutter.settings.FlutterSettings;
+  import io.flutter.utils.VmServiceListenerAdapter;
+  import org.dartlang.vm.service.VmService;
+  import org.dartlang.vm.service.element.Event;
+  import org.jetbrains.annotations.NotNull;
+  import org.jetbrains.annotations.Nullable;
+
+  import javax.swing.*;
+  import java.util.ArrayList;
+  import java.util.HashMap;
+  import java.util.Map;
+  import java.util.Objects;
+
+  // TODO(devoncarew): Display an fps graph.
+
+  // TODO(devoncarew): Ensure all actions in this class send analytics.
```

Comment:
note: we can delete this todo now

Reply:
removed this TODO but not the one above it.